### PR TITLE
Asciidoctor documentation

### PR DESCRIPTION
**Pull request by mariogarcia**
Hi:

First of all, thanks for such useful library :)

I've been using your library for a while, and I decided to start a documentation in order to understand better how the library works.

This PR is splitted in several commits:
- (1) Add Groovydoc to all methods at `spock.genesis.Gen`
- (2) Add asciidoctor documentation using pre-existent examples in `SampleSpec.groovy`
- (3) Add some entries to .gitignore
- (4) Add more examples about non composite generators to `SamplesSpec.groovy` in order to keep completing asciidoctor docs
- (5) Add more examples to `SamplesSpec.groovy` to add an asciidoctor section about combining generators
- (6) Add gradle plugin to publish asciidoctor manual to project's `gh-pages` using `circleci`

In general points from `1` through `5` are pretty straight forward. In order to generate the documentation you only have to execute:

```
./gradlew alldocs
```

and check `build/asciidoc/html5/index.html` and review the documentation. Please notice that `alldocs` also copy `groovydoc` to the asciidoctor folder in order to make it available through the `gh-pages` asciidoctor documentation.

The last point `(6)` is about publishing the documentation automatically to the project's `gh-pages`. Documentation will be published using the https://github.com/ajoberstar/gradle-git gradle plugin.

This step depends on you adding a github personal token to `circleci` the same way I guess you added your bintray credentials. In the following link is explained for TravisCI but I think it should work the same for `circleci` https://github.com/ajoberstar/gradle-git/wiki/Travis-CI-Authentication

Basically you should add a variable named `GH_TOKEN` with the personal token retrieved from github to your `circleci` environment.

While writing the documentation I've found a bug, but I rather want to see what happens with this PR before sending you anything else.

Again thanks for your library and I hope you like this contribution.

Mario

### CODE REVIEW

**Comment by Bijnagte** [comment]
@mariogarcia Thanks for all your work on this! i just read through everything and it looks good. I will pull down your branch and do a build so I can check out the docs before hitting the merge button. Sorry it took a couple of days for me to get to this.

**Comment by Bijnagte** [comment]
http://bijnagte.github.io/spock-genesis/

**Comment by mariogarcia** [comment]
Awesome :-)

No worries, I fully understand sometimes is hard to save time. It happens to me too.

BTW thanks for adding me as a contributor to the repository.
